analysis/online/R3BOnlineSyncCheck.cxx

```diff
@@ -0,0 +1,164 @@
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "R3BOnlineSyncCheck.h"
+#include "R3BEventHeader.h"
+#include "R3BLogger.h"
+#include "R3BSyncCheckData.h"
+
+#include "FairLogger.h"
+#include "FairRootManager.h"
+#include "FairRunAna.h"
+#include "FairRunOnline.h"
+#include "FairRuntimeDb.h"
+#include "TCanvas.h"
+#include "TClonesArray.h"
+#include "TFolder.h"
+#include "TH2F.h"
+#include "THttpServer.h"
+
+R3BOnlineSyncCheck::R3BOnlineSyncCheck()
+    : R3BOnlineSyncCheck("OnlineSyncCheck", 1)
+{
+}
+
+R3BOnlineSyncCheck::R3BOnlineSyncCheck(const TString& name, Int_t iVerbose)
+    : FairTask(name, iVerbose)
+    , fCA(nullptr)
+    , header(nullptr)
+    , fTpat(-1)
+    , fTrig(-1)
+    , canvas(nullptr)
+{
+}
+
+TString R3BOnlineSyncCheck::EnumName(int value)
+{
+    switch (value)
+    {
+        case MASTER_SC:
+            return { TString("MASTER") };
+        case MASTERREF_SC:
+            return { TString("MASTER_REF") };
+        case MUSIC_SC:
+            return { TString("MUSIC") };
+        case RPC_SC:
+            return { TString("RPC") };
+        case S2_SC:
+            return { TString("S2") };
+        case FOOT1_SC:
+            return { TString("FOOT1") };
+        case FOOT2_SC:
+            return { TString("FOOT2") };
+        default:
+            return { TString("Invalid enum value") };
+    }
+}
+
+InitStatus R3BOnlineSyncCheck::Init()
+{
+    R3BLOG(info, "");
+    FairRootManager* mgr = FairRootManager::Instance();
+    R3BLOG_IF(fatal, nullptr == mgr, "FairRootManager not found");
+
+    FairRunOnline* run = FairRunOnline::Instance();
+    run->GetHttpServer()->Register("", this);
+
+    header = dynamic_cast<R3BEventHeader*>(mgr->GetObject("EventHeader."));
+    R3BLOG_IF(fatal, !header, "Branch EventHeader. not found");
+
+    fCA = dynamic_cast<TClonesArray*>(mgr->GetObject("SyncCheckData"));
+    R3BLOG_IF(fatal, !fCA, "Branch SyncCheckData not found");
+
+    canvas = new TCanvas("Sync check", "Sync check");
+    canvas->Divide(NXPADS, NYPADS);
+
+    for (int i = 0; i < NUM_SC; ++i)
+    {
+        canvas->cd(i + 1);
+        TString hname = EnumName(i);
+        auto* fh2 = new TH2F(hname.Data(), hname.Data(), XBINS, XMIN, XMAX, YBINS, YMIN, YMAX);

```

Comment:
Please, use here the new definition of histograms:
auto* fh2 = R3B::root_owned<TH2F>(hname.Data(), hname.Data(), XBINS, XMIN, XMAX, YBINS, YMIN, YMAX);

Reply:
sure, I will do it. In the meantime I have committed some other updates for the reader and online histograms

Reply:
I think it should be replaced like this:

auto* fh2 = R3B::root_owned<TH2F>(hname.Data(), hname.Data(), XBINS, XMIN, XMAX, YBINS, YMIN, YMAX);

this also requires inclusion of the header:

#include"R3BShared.h"

Reply:
Yes, it requires --> #include "R3BShared.h"
and for the definition of histograms, it was commented by @YanzhaoW in the PR #830

Reply:
ok, I think it is ready now. There are still a few minor clang-tidy warnings but I think they can be safely ignored. Let me know if I should modify anything else, otherwise this can be merged